adapters/tlsio_openssl.c

```diff
@@ -1036,6 +1037,7 @@ static int create_openssl_instance(TLS_IO_INSTANCE* tlsInstance)
                     }
 
                     tlsInstance->ssl = SSL_new(tlsInstance->ssl_context);
+
                     if (tlsInstance->ssl == NULL)
                     {
                         (void)BIO_free(tlsInstance->in_bio);
```

Comment:
Looks like a memory leak would happen here if SSL_new fail. Need to investigate.

Reply:
`out_bio`?